### DISCUSSION TITLE
Modify by pytest_itemcollected item._nodeid， setup step and teardown step in the report are missing

### DISCUSSION THREAD

**Discussion by muzijian07king** [pytest-dev/pytest#10022] [Q&A] [answered]
#Modify by pytest_itemcollected item._nodeid
def pytest_itemcollected(item):
    name = get_runner_app()['software'].replace(' ', '-')
    item._nodeid = name + '_' + item._nodeid

![image](https://user-images.githubusercontent.com/71326593/172125015-797ff898-4ca5-4777-8076-7f056febb191.png)
![image](https://user-images.githubusercontent.com/71326593/172125234-8eea17ff-04de-48fe-b727-9a5a5029009e.png)

**Comment by muzijian07king** [comment]
pytest  7.0.1
allure-pytest 2.8.6

**Reply by nicoddemus** [reply]
Changing the `.nodeid` attribute is not supported, that's the reason why the setup/teardown steps are missing I believe.

**Comment by RonnyPfannschmidt** [comment, marked as answer]
this breaks all expectations for pytest, its absolutely not allowed to do this, just stop

**Reply by muzijian07king** [reply]
thank you
